src/sentry/rules/actions/utils.py

```diff
@@ -0,0 +1,96 @@
+from collections import defaultdict
+from typing import Any
+
+from sentry.api.serializers.models.rule import generate_rule_label
+from sentry.models.environment import Environment
+from sentry.models.rule import Rule
+
+
+def get_changed_data(
+    rule: Rule, rule_data: dict[str, list[Any]], rule_data_before: dict[str, list[Any]]
+) -> dict[str, list[Any]]:
+    """
+    Generate a list per type of issue alert rule data of what changes occurred on edit.
+    """
+    changed_data = defaultdict(list)
+
+    for condition in rule_data.get("conditions", []):
+        if condition not in rule_data_before.get("conditions", []):
+            label = generate_rule_label(rule_data.get("project"), rule, condition)
+            changed_data[condition["id"]].append(f"Added condition '{label}'")
```

Comment:
nit: This is more personal preference I guess, but for iterating I think it could be easier to use set diffs, like:
```Python
old_conditions = set(rule_data_before.get("conditions", []))
new_conditions = set(rule_data.get("conditions", []))

for condition in new_conditions - old conditions:
    ...
```

Reply:
unfortunately this can't work since conditions and actions are a list of dictionaries, and sets can only have hashable stuff